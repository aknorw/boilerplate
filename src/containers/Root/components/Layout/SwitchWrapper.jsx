import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Switch } from 'react-router-dom'

const SwitchStyle = styled.main.attrs({
  role: 'main',
})`
  flex: 1;
`

const SwitchWrapper = ({ children }) => (
  <SwitchStyle>
    <Switch>
      {children}
    </Switch>
  </SwitchStyle>
)

SwitchWrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
}

SwitchWrapper.defaultProps = {
  children: null,
}

export default SwitchWrapper
