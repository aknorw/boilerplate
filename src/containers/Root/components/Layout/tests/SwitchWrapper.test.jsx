import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import SwitchWrapper from '../SwitchWrapper'

describe('<SwitchWrapper />', () => {
  it('should be defined', () => {
    expect(SwitchWrapper).toBeDefined()
  })
  it('should render correctly', () => {
    const tree = shallow(
      <SwitchWrapper />,
    )
    expect(tree).toMatchSnapshot()
  })
  it('should have style rules', () => {
    const tree = shallow(
      <SwitchWrapper />,
    )
    expect(tree).toHaveStyleRule('flex', '1')
  })
})
