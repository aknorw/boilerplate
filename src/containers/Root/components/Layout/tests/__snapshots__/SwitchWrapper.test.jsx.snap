// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchWrapper /> should render correctly 1`] = `
<styled.main>
  <Switch />
</styled.main>
`;
